File: 952-word-subsets/word-subsets.java
class Solution {
    public List<String> wordSubsets(String[] words1, String[] words2) {
        int[] words2Count = new int[26];
        for (String word : words2) {
            int[] tempCount = new int[26];
            for (char c : word.toCharArray()) {
                tempCount[c - 'a']++;
            }
            for (int i = 0; i < 26; i++) {
                words2Count[i] = Math.max(words2Count[i], tempCount[i]);
            }
        }

        List<String> result = new ArrayList<>();
        for (String word : words1) {
            int[] wordCount = new int[26];
            for (char c : word.toCharArray()) {
                wordCount[c - 'a']++;
            }
            boolean isUniversal = true;
            for (int i = 0; i < 26; i++) {
                if (wordCount[i] < words2Count[i]) {
                    isUniversal = false;
                    break;
                }
            }
            if (isUniversal) {
                result.add(word);
            }
        }
        System.out.println(result);
        return result;
    }
}

File: 952-word-subsets/word-subsets.py
class Solution:
    def wordSubsets(self, words1: List[str], words2: List[str]) -> List[str]:
        words2_count = [0] * 26
        for word in words2:
            temp_count = [0] * 26
            for char in word:
                temp_count[ord(char) - ord('a')] += 1
            for i in range(26):
                words2_count[i] = max(words2_count[i], temp_count[i])
        result = []
        for word in words1:
            word_count = [0] * 26
            for char in word:
                word_count[ord(char) - ord('a')] += 1
            is_universal = True
            for i in range(26):
                if word_count[i] < words2_count[i]:
                    is_universal = False
                    break
            if is_universal:
                result.append(word)

        return result